File: src/main/java/howtoprogram/austen/Histogram.java
package howtoprogram.austen;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;

/**
 * Collects word usage data.
 */
public class Histogram {

    private Map<String,Integer> counter = new HashMap<>();

    public void record(String word) {
        Integer currentCount;
        if (counter.containsKey(word)) {
            currentCount = counter.get(word);
        } else {
            currentCount = 0;
        }
        Integer newCount = currentCount + 1;
        counter.put(word, newCount);
    }

    public Set<String> allWords() {
        return counter.keySet();
    }

    public int numberOfTimesGiven(String word) {
        if (counter.containsKey(word)) {
            return counter.get(word);
        } else {
            return 0;
        }
    }
}

File: src/main/kotlin/pfb/austen/Histogram.kt
package pfb.austen

/**
 * Collects word usage data.
 */
class Histogram {

    val counter = mutableMapOf<String, Int>()

    fun record(word: String) {
        val currentCount = counter[word] ?: 0
        val newCount = currentCount + 1
        counter.put(word, newCount)
    }

    fun allWords(): Set<String> {
        return counter.keys
    }

    fun numberOfTimesGiven(word: String): Int {
        return counter[word] ?: 0
    }
}